#2925 text separator for concatenating errors/main

Freelancingonupwork opened the pull request:
By default TextSeparator will be " "  for concatenating errors. and someone can set this as well.

Fixes #2925 
![TextSeparator](https://github.com/MarimerLLC/csla/assets/56026691/524985de-39b9-460f-a665-d2863b43c89f)

Freelancingonupwork commented:
> By default TextSeparator will be " " for concatenating errors. and someone can set this as well.

I see some issue in PR  I see commits which are already merged in main branch please don't review until I fixed it

rockfordlhotka commented:
@Freelancingonupwork I switched the PR to a draft. You can switch it back once you are ready.

Freelancingonupwork commented:
> @Freelancingonupwork I switched the PR to a draft. You can switch it back once you are ready.

HI @rockfordlhotka,  I fixed the issues by merging main into my feature branch now review the PR

rockfordlhotka commented on Source/Csla/Csla.csproj:
```diff
@@ -12,6 +12,7 @@
     <BaseOutputPath>..\..\Bin</BaseOutputPath>
     <GenerateDocumentationFile>True</GenerateDocumentationFile>
     <GeneratePackageOnBuild>False</GeneratePackageOnBuild>
+    <Version>9.0.0.5</Version>
```
There should be no `Version` element here - it comes from Directory.build.props

Freelancingonupwork replied:
I will remove it

rockfordlhotka commented on Source/Csla.Blazor/PropertyInfo.cs:
```diff
@@ -16,8 +16,10 @@ namespace Csla.Blazor
+    /// <summary>
+    /// Gets the value of the TextSeparator
+    /// </summary>
+    public string TextSeparator { get; }
```
Why are we changing the default from ", " to " "? I would think we should keep the default the same?

Freelancingonupwork replied:
it was there in description that Default will be " ".

![image](https://github.com/MarimerLLC/csla/assets/56026691/fab37e14-ce17-49c2-9752-61327f82e3f8)

rockfordlhotka replied:
My bad 🫢

rockfordlhotka commented on Source/Csla.Blazor/PropertyInfo.cs:
```diff
@@ -28,14 +30,16 @@ public class PropertyInfo : IPropertyInfo, INotifyPropertyChanged
     /// <param name="propertyName">Property name</param>
-    public PropertyInfo(object model, string propertyName)
+    /// <param name="textSeprator">Text Seprator</param>
+    public PropertyInfo(object model, string propertyName, string textSeprator = " ")
```
Would this be better as a optional parameter so people who want to use the default can just ignore this new parameter?

rockfordlhotka commented on Source/Csla.Blazor/ViewModel.cs:
```diff
@@ -387,7 +387,24 @@ public IPropertyInfo GetPropertyInfo<P>(Expression<Func<P>> property)
       var keyName = property.GetKey();
       var identifier = Microsoft.AspNetCore.Components.Forms.FieldIdentifier.Create(property);
-      return GetPropertyInfo(keyName, identifier.Model, identifier.FieldName);
+      return GetPropertyInfo(keyName, identifier.Model, identifier.FieldName, " ");
```
Shouldn't this use the default rather than hard-coding the separator? Same for other instances below.

Freelancingonupwork replied:
Yes, you are correct. i will do that.
